ANMN/burst_averaged_product/burstproduct/burst_average_product.py

```diff
@@ -0,0 +1,401 @@
+        [var_values, var_qc_flag_exclusion] = get_var_val_var_qc_exclusion(netcdf_file_path, varname, min_bad_qc_flag(varname))
+        [time_mean_burst, var_mean_burst, var_min_burst, var_max_burst, var_std_burst, var_num_obs] = burst_average_data(time_values, var_values, burst_duration, var_qc_flag_exclusion)
+
+        burst_vars[varname] = [time_mean_burst, var_mean_burst, var_min_burst, var_max_burst, var_std_burst, var_num_obs]
+
+    burst_vars = clean_burst_vars(burst_vars)
+    return burst_vars
+
+def clean_burst_vars(burst_vars):
+    """
+    For every burst var created, look for the first index of non NaN value. The
+    lower value will be the one kept, and the new start index of each burst variable
+    including the TIME.
+    """
+    min_index = None
+    for var in burst_vars.keys():
+        var_mean_burst = burst_vars[var][1] # first non TIME product
+        min_index_var  = next(x for x, y in enumerate(var_mean_burst) if not isnan(y))
+
+        if min_index == None:
+            min_index = min_index_var
+        else:
+            min_index = min(min_index, min_index_var)
+
+    # remove the first common NaN to all variables
+    for var in burst_vars.keys():
+        for product in range(len(burst_vars[var])):
+            burst_vars[var][product] = burst_vars[var][product][min_index:-1]
+
+    # look for last non NaN. we just have to reverse the list
+    max_non_nan_idx = None
+    for var in burst_vars.keys():
+        var_mean_burst       = burst_vars[var][1] # first non TIME product
+        var_mean_burst       = pd.Series(var_mean_burst) # create a pandas series to find easily last non nan index
+        max_last_non_nan_var = var_mean_burst.last_valid_index()
+        if max_non_nan_idx == None:
+            max_non_nan_idx = max_last_non_nan_var
+        else:
+            max_non_nan_idx = max(max_non_nan_idx, max_last_non_nan_var)
+
+    # remove the last common NaN to all variables
+    for var in burst_vars.keys():
+        for product in range(len(burst_vars[var])):
+            burst_vars[var][product] = burst_vars[var][product][:max_non_nan_idx]
+
+    return burst_vars
+
+def get_time_val(netcdf_file_path):
+    """
+    return the TIME numeric values from a NetCDF file
+    """
+    netcdf_file_obj = Dataset(netcdf_file_path, 'r')
+    time            = netcdf_file_obj.variables['TIME']
+    time_values     = time[:]
+    netcdf_file_obj.close()
+    return time_values
+
+def get_var_val_var_qc_exclusion(netcdf_file_path, varname, qc_flag):
+    """
+    for a qc flag values [0:9], returns the var_values array, and var_qc_flag_exclusion
+    which is a boolean array returning True for all qc >= qc_flag
+
+    Also exlude all TIME values outside of deployment range as defined in gatts
+    """
+    netcdf_file_obj       = Dataset(netcdf_file_path, 'r')
+    shape_var_array = netcdf_file_obj.variables[varname].shape
+
+    if len(shape_var_array) == 3:
+        # WQM files
```

Comment:
This is not a differene between WQM and CTD files. It depends on which version of the Toolbox processed it.